### PR TITLE
Adds Experiment Manager and executor service

### PR DESCRIPTION
**Pull request by bharathappali**
Adds Experiment Manager initiation, creates the executor service, adds settings and constants which are used in the EM

Signed-off-by: bharathappali <abharath@redhat.com>

### CODE REVIEW

**Comment by bharathappali** [comment]
- Updated the branch with calculation for executors
- null is placed left to the comparison
- max pool size and the current pool size is checked with the limits placed in EM settings class
- Renamed EMS to EMSettings
- Renamed `launch` with `start`

@dinogun Can I please have this one reviewed?

**Comment by bharathappali** [comment]
- Added constant for Minimum Pool Size check `MIN_EXECUTOR_POOL_SIZE`
- Added definitions for the Multipiers via comments

@dinogun This PR is now ready to be re-reviewed. Thanks in advance.

File: src/main/java/com/autotune/experimentManager/utils/EMConstants.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package com.autotune.experimentManager.utils;
+
+public class EMConstants {
+    public static class APIPaths {
+        public static final String ROOT = "/";
```
Comment (dinogun) [inline src/main/java/com/autotune/experimentManager/utils/EMConstants.java:21]: Why do we need this here ?

File: src/main/java/com/autotune/experimentManager/utils/EMConstants.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    public static class DeploymentStrategies {
+        public static String ROLLING_UPDATE = "rollingUpdate";
+        public static String NEW_DEPLOYMENT = "newDeployment";
+    }
+
+    public static class Logs {
+        public static class LoggerSettings {
+            public static String DEFUALT_LOG_LEVEL = "ALL";
```
Comment (dinogun) [inline src/main/java/com/autotune/experimentManager/utils/EMConstants.java:51]: All I see it is misspelled here too

File: src/main/java/com/autotune/experimentManager/utils/EMConstants.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            public static String CREATE_SCHEDULED_EXECUTOR = "Creating scheduled executor";
+            public static String START_EXECUTE_TRIAL = "Starting to execute a trial";
+            public static String START_SCHEDULED_EXECUTE_TRIAL = "Starting to execute a scheduled trial";
+            public static String START_STAGE_PROCESSORS = "Starting stage processors";
+        }
+    }
+
+    public static class InputJsonKeys {
+        public static class GetTrailStatusInputKeys {
+            public static String RUN_ID ="runId";
```
Comment (dinogun) [inline src/main/java/com/autotune/experimentManager/utils/EMConstants.java:75]: Space after `=`

**Review by dinogun — CHANGES_REQUESTED** [review]
Needs some minor changes

File: src/main/java/com/autotune/experimentManager/ExperimentManager.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2021 Red Hat, IBM Corporation and others.
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply src/main/java/com/autotune/experimentManager/ExperimentManager.java]: Thanks for the review. I have made changes in all the files of this patch.

File: src/main/java/com/autotune/experimentManager/ExperimentManager.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    public static EMExecutorService emExecutorService;
+
+    public static void initializeEM() {
+        // Initializes the executor services needed by the EM
+        emExecutorService = EMExecutorService.getInstance();
+    }
+
+    public static void start(ServletContextHandler contextHandler) {
+        // Launches / starts the EM
+        Configurator.setAllLevels(LogManager.getRootLogger().getName(), Level.toLevel(EMConstants.Logs.LoggerSettings.DEFUALT_LOG_LEVEL));
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply src/main/java/com/autotune/experimentManager/ExperimentManager.java]: Oops! missed it out somehow earlier, changed it now to `DEFAULT`. I feel it shouldn't face any compilation issue as it's a variable name.

File: src/main/java/com/autotune/experimentManager/core/EMExecutorService.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2021 Red Hat, IBM Corporation and others.
```
[the remark this answers is not in the document]
Comment (bharathappali) [reply src/main/java/com/autotune/experimentManager/core/EMExecutorService.java]: Done

File: src/main/java/com/autotune/experimentManager/utils/EMConstants.java (hunk 1 above)
Comment (bharathappali) [reply src/main/java/com/autotune/experimentManager/utils/EMConstants.java]: Replaced it with `ServerContext.ROOT_CONTEXT` from DA

File: src/main/java/com/autotune/experimentManager/utils/EMConstants.java (hunk 2 above)
Comment (bharathappali) [reply src/main/java/com/autotune/experimentManager/utils/EMConstants.java]: Corrected it over here.

File: src/main/java/com/autotune/experimentManager/utils/EMConstants.java (hunk 3 above)
Comment (bharathappali) [reply src/main/java/com/autotune/experimentManager/utils/EMConstants.java]: Made changes to add space.

**Comment by bharathappali** [comment]
Thanks, @dinogun for the review. I have made the changes mentioned.

**Pull request merged**